event_session/models/event_session.py

```diff
@@ -0,0 +1,524 @@
+        now = fields.Datetime.now()
+        for rec in self:
+            rec.event_registrations_open = (
+                rec.event_registrations_started
+                and (not rec.date_end or rec.date_end >= now)
+                and (not rec.seats_limited or rec.seats_available)
+                and (
+                    not rec.event_ticket_ids
+                    or any(ticket.sale_available for ticket in rec.event_ticket_ids)
+                )
+            )
+
+    @api.depends(
+        "event_ticket_ids.seats_available",
+        "seats_limited",
+        "seats_available",
+    )
+    def _compute_event_registrations_sold_out(self):
+        """Similar to core's :meth:`event_event._compute_event_registrations_sold_out`"""
+        for rec in self:
+            if rec.seats_limited and not rec.seats_available:
+                rec.event_registrations_sold_out = True
+            elif rec.event_ticket_ids:
+                rec.event_registrations_sold_out = not any(
+                    ticket.seats_available > 0 if ticket.seats_limited else True
+                    for ticket in rec.event_ticket_ids
+                )
+            else:
+                rec.event_registrations_sold_out = False
+
+    @api.depends("event_id.event_mail_ids")
+    def _compute_event_mail_ids(self):
+        """Compute event mail ids from its parent event
+
+        The email schedulers for sessions are used to track their independent states,
+        but the management is done directly from the parent event.event.
+
+        This method takes care of synchronizing the session's schedulers with those
+        of their parent events.
+        """
+        for rec in self:
+            existing_schedulers = rec.event_mail_ids.scheduler_id
+            event_schedulers = rec.event_id.event_mail_ids
+            # Unlink the ones no-longer in sync
+            to_unlink = rec.event_mail_ids.filtered(
+                lambda r: r.scheduler_id not in event_schedulers
+            )
+            if to_unlink:
+                rec.event_mail_ids = [
+                    fields.Command.unlink(scheduler.id) for scheduler in to_unlink
+                ]
+            # Create missing ones
+            to_create = event_schedulers - existing_schedulers
+            if to_create:
+                rec.event_mail_ids = [
+                    fields.Command.create(
+                        scheduler._prepare_session_mail_scheduler_vals(rec)
+                    )
+                    for scheduler in to_create
+                ]
+                # Force recomputation of scheduled date
+                rec.event_mail_ids._compute_scheduled_date()
+
+    @api.model
+    def _read_group_stage_ids(self, stages, domain, order):  # pragma: no cover
+        return self.env["event.event"]._read_group_stage_ids(stages, domain, order)
+
+    @api.constrains("seats_max", "seats_available", "seats_limited")
+    def _check_seats_limit(self):
```

Comment:
Could you update this method so that it's named the same as, and behaves the same as, [the `_check_seats_availability` on `event.event`](https://github.com/odoo/odoo/blob/d44106309c4a11fc3c0dca4ed8457e1f27bd1527/addons/event/models/event_event.py#L555) ?

The general idea is the `event.session` should act almost exactly as the core `event.event` for regular events. So most of the methods, fields and constrains here are adaptations of what you find in `event.event`

Reply:
> The general idea is the event.session should act almost exactly as the core event.event for regular events. So most of the methods, fields and constrains here are adaptations of what you find in event.event

It'd be great if you can check the other fields / methods with this in mind, too 🙏🏻